### PR TITLE
drivers: sensor: explorir_m: initial implementation

### PR DESCRIPTION
**Pull request by jeppenodgaard**
The ExplorIR®-M is a small footprint low power NDIR CO2 sensor.
The initial driver uses polling mode, is able to fetch the filtered co2 value and supports set/get a digital filter on the sensor.

### CODE REVIEW

**Review by bjarki-andreasen — DISMISSED** [review]
Looks good :)

File: samples/sensor/explorir_m/src/main.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	const struct device *const dev = DEVICE_DT_GET_ANY(gss_explorir_m);
+
+	if (!device_is_ready(dev)) {
+		LOG_ERR("%s is not ready", dev->name);
+		return 0;
+	}
+
+	while (1) {
+		if (sensor_sample_fetch(dev) == 0 &&
+		    sensor_channel_get(dev, SENSOR_CHAN_CO2, &value) == 0) {
```
Comment (MaureenHelm) [inline samples/sensor/explorir_m/src/main.c:25]: Please convert this sample application into a more generic one that can be used with any CO2 sensor. See `samples/sensor/accel_polling` for example

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: samples/sensor/explorir_m/src/main.c (hunk 1 above)
Comment (jeppenodgaard) [reply samples/sensor/explorir_m/src/main.c]: Good idea. I've changed the sample.

An even more generic polling sample might be useful which can be used with any type of sensor.
Then maybe only a dts overlay would be required. Not sure how to map which channel is supported by the sensor though.

File: samples/sensor/co2_polling/prj.conf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CONFIG_STDOUT_CONSOLE=y
+CONFIG_LOG=y
+CONFIG_SHELL=y
+CONFIG_SENSOR_SHELL=y
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
```
Comment (MaureenHelm) [inline samples/sensor/co2_polling/prj.conf:6]: These should get selected by the sensor driver

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Can you also update the commit message to reflect the change to a generic c02 sensor sample?
```
samples: sensors: add explorir_m co2 sensor

Add sample for ExplorIR-M CO2 sensor.

Signed-off-by: Jeppe Odgaard <jeppe.odgaard@prevas.dk>
```

**Comment by jeppenodgaard** [comment]
Thanks @MaureenHelm. The commit message is updated and `CONFIG_UART_INTERRUPT_DRIVEN` has been moved to the board config.

**Review by MaureenHelm — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Pull request merged**